controllers/host/host_controller.go

```diff
@@ -1173,7 +1173,7 @@ func (r *HostReconciler) ReconcileHostByState(
 		}
 
 		// Clean up strategy required after Disabled and enabled attributes are all in-sync
-		if strategy_required {
+		if instance.Status.StrategyRequired == cloudManager.StrategyUnlockRequired && !r.CloudManager.GetStrategySent() {
```

Comment:
Any reason to check if the strategy was sent and why the "LockRequired" was moved out of the scope?
Worrying about the cases, e.g.:
Strategy failed to be built, and the host was lock required. If we want to change the configuration back to what was previously insync, will the strategy lock required be stuck in this state?

Need also update the comment above to explain why we are not clearing up the strategy status even in-sync

Reply:
Reverted above condition with slight improvement so that it can handle corner cases where strategy is in progress or strategy creation request yet to be sent.